src/polygon/src/polygon.cpp

```diff
@@ -838,9 +838,10 @@ class Monotones {
     }
 #if MANIFOLD_PAR == 'T' && TBB_INTERFACE_VERSION >= 10000 && \
     __has_include(<pstl/glue_execution_defs.h>)
-    std::sort(std::execution::par_unseq, starts.begin(), starts.end(), cmp);
+    std::stable_sort(std::execution::par_unseq, starts.begin(), starts.end(),
+                     cmp);
 #else
-    std::sort(starts.begin(), starts.end(), cmp);
+    std::stable_sort(starts.begin(), starts.end(), cmp);
```

Comment:
Do these stable sorts make much difference?

Reply:
probably not, this is just in case. the performance of stable sort is not a lot different than sort from my testing

(FYI: rust uses stable sort by default, and the c++ sort is named unstable sort in rust and we will only use it when profile shows it is beneficial)